x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/FIPS140LicenseBootstrapCheck.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security;
+
+import org.elasticsearch.bootstrap.BootstrapCheck;
+import org.elasticsearch.bootstrap.BootstrapContext;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.LicenseService;
+
+import java.util.EnumSet;
+
+/**
+ * A bootstrap check which enforces the licensing of FIPS
+ */
+final class FIPS140LicenseBootstrapCheck implements BootstrapCheck {
+
+    static final EnumSet<License.OperationMode> ALLOWED_LICENSE_OPERATION_MODES =
+        EnumSet.of(License.OperationMode.PLATINUM, License.OperationMode.TRIAL);
+
+    private final boolean isInFipsMode;
+
+    FIPS140LicenseBootstrapCheck(boolean isInFipsMode) {
+        this.isInFipsMode = isInFipsMode;
+    }
+
+    @Override
+    public BootstrapCheckResult check(BootstrapContext context) {
+        if (isInFipsMode) {
+            License license = LicenseService.getLicense(context.metaData);
+            if (license != null && ALLOWED_LICENSE_OPERATION_MODES.contains(license.operationMode()) == false) {
+                return BootstrapCheckResult.failure("FIPS mode is only allowed with a Platinum or Trial license");
+            }
+        }
+        return BootstrapCheckResult.success();
+    }
+
+    @Override
+    public boolean alwaysEnforce() {
+        return isInFipsMode;
```

Comment:
This can be `true` since the check is itself guarded by `isInFipsMode`.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/32437/commits/2ab94511a742abb346a609c063bf257e46ff0584